src/importer/features/main/index.tsx

```diff
@@ -196,6 +201,18 @@ export default function Main(props: CSVImporterProps) {
             onSuccess={(columnMapping) => {
               setIsSubmitting(true);
               setColumnMapping(columnMapping);
+              if (onHeadersMapped) {
+                onHeadersMapped(selectedHeaderRow, columnMapping, originalFile)
+                  .then(() => {
+                    setIsSubmitting(false);
+                    goNext();
+                  })
+                  .catch((error) => {
+                    console.error("onHeadersMapped error", error);
+                    setDataError("An error occurred while processing the data");
+                  });
+                return;
+              }
```

Comment:
I'm getting this error in testing when `onHeadersMapped` isn't set: `Uncaught TypeError: Cannot read properties of undefined (reading 'then')`

Maybe update to use `Promise.resolve`?

```suggestion
              if (onHeadersMapped) {
                Promise.resolve(onHeadersMapped(selectedHeaderRow, columnMapping, originalFile))
                  .then(() => {
                    setIsSubmitting(false);
                    goNext();
                  })
                  .catch((error) => {
                    console.error("onHeadersMapped error", error);
                    setDataError("An error occurred while processing the data");
                  });
                return;
              }
```

Reply:
@ciminelli 
This code block is executed only when `onHeadersMapped` is defined

```
if (onHeadersMapped) {
...
}
```

If you could upload video of how/where you are getting this error, it would be helpful.